app/controllers/health_check_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class HealthCheckController < ApplicationController
+  def all
+    response = 'success'
+    @cache_expire = 10.seconds
+
+    begin
+      cache_check
+      database_check
+    rescue StandardError => e
+      response = "Health Check Failure: #{e}"
```

Comment:
I would make this the exit point directly, less spagety
and remove the initialization at the top and  just send the constant at the bottom. Plain and simple

Reply:
Wouldn't that require 2 separate return statements?

Reply:
Yes, that is correct. The single point of exit recommendation is long gone in favour of simplicity and readability, that is also how you get rid of nedted if and else statements